parlai/chat_service/services/messenger/agents.py

```diff
@@ -81,13 +79,17 @@ def put_data(self, message):
                 'episode_done': False,
                 'text': text,
                 'id': self.disp_id,
-                'report_sender': message['message'].get('report_sender', None),
-                'sticker_sender': message.get('sticker_sender', None),
+                'report_sender': message.get('report_sender', None),
+                'sticker_sender': messenger_message.get('sticker_sender', None),
                 'img_attempt': img_attempt,
             }
             if img_attempt and self.data.get('allow_images', False):
-                action['image_url'] = message['message'].get('image_url')
-                action['attachment_url'] = message['message'].get('attachment_url')
+                action['image_url'] = message.get('image_url')
+                action['attachment_url'] = message.get('attachment_url')
+                if action['image_url'] is None and 'attachments' in message:
```

Comment:
i believe you don't need to check for `'attachments' in message` here because `image_attempt` already checks for that